strongbox-commons/src/main/java/org/carlspring/strongbox/booters/SystemPropertiesInitializer.java

```diff
@@ -0,0 +1,56 @@
+package org.carlspring.strongbox.booters;
+
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.core.env.ConfigurableEnvironment;
+
+/**
+ * Ensures that a handful of core system properties are set prior to starting the application.
+ *
+ * @author cbono
+ */
+class SystemPropertiesInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext>
+{
+    // All of these properties are package-protected so VisibleForTesting
+    static final String STRONGBOX_HOME_PROPERTY_NAME = "strongbox.home";
+    static final String STRONGBOX_VAULT_PROPERTY_NAME = "strongbox.vault";
+    static final String STRONGBOX_STORAGE_BASEDIR_PROPERTY_NAME = "strongbox.storage.booter.basedir";
+    static final String LOGGING_DIR_PROPERTY_NAME = "logging.dir";
+    static final String LOGGING_CONFIG_FILE_PROPERTY_NAME = "logging.config.file";
+    static final String JAVA_TMP_DIR_PROPERTY_NAME = "java.io.tmpdir";
+    static final String EHCACHE_DISK_DIR_PROPERTY_NAME = "ehcache.disk.store.dir";
+
+    @Override
+    public void initialize(ConfigurableApplicationContext applicationContext)
+    {
+        final ConfigurableEnvironment env = applicationContext.getEnvironment();
+        final String homeDir = env.getRequiredProperty(STRONGBOX_HOME_PROPERTY_NAME);
+        final String vaultDir = env.getRequiredProperty(STRONGBOX_VAULT_PROPERTY_NAME);
+        final String storageBooterBaseDir = env.getRequiredProperty(STRONGBOX_STORAGE_BASEDIR_PROPERTY_NAME);
+
+        if (System.getProperty(LOGGING_DIR_PROPERTY_NAME) == null)
+        {
+            System.setProperty(LOGGING_DIR_PROPERTY_NAME, vaultDir + "/logs");
+        }
+
+        if (System.getProperty(LOGGING_CONFIG_FILE_PROPERTY_NAME) == null)
+        {
+            System.setProperty(LOGGING_CONFIG_FILE_PROPERTY_NAME, homeDir + "/etc/logback-spring.xml");
+        }
+
+        if (System.getProperty(JAVA_TMP_DIR_PROPERTY_NAME) == null)
+        {
+            System.setProperty(JAVA_TMP_DIR_PROPERTY_NAME, vaultDir + "/tmp");
+        }
+
+        if (System.getProperty(EHCACHE_DISK_DIR_PROPERTY_NAME) == null)
+        {
+            System.setProperty(EHCACHE_DISK_DIR_PROPERTY_NAME, homeDir + "/cache");
+        }
+
+        if (System.getProperty(STRONGBOX_STORAGE_BASEDIR_PROPERTY_NAME) == null)
```

Comment:
the System.getProperty(`strongbox.storage.booter.basedir`) seems never used directly, so we don't need this.